### PR TITLE
Support a `.well-known` www directory for external CA validation

### LINKED ISSUE #2602: Allow apache/apparmor file exception for proving onion ownership to a CA
# Feature request

## Description

Our [documentation](https://docs.securedrop.org/en/stable/https_source_interface.html) does a great job at explaining the process for obtaining and adding an HTTPS cert to the source interface. Unfortunately we are missing logic and direction on how an admin can verify their instance with a CA that attempts to prove onion server ownership. Usually they will ask an admin to drop a file to prove ownership.

We should update apparmor / apache / and documentation to better guide administrators on how they can address this scenario without asking them to muck around with custom temporary configuration.

## User Stories

As an administrator, I want to be able to add files easily to securedrop to prove onion domain control to an external CA.

### PR DESCRIPTION
**Pull request by msheiny**
## Status

Ready for review

## Description of Changes

Fixes #2602

Changes proposed in this pull request:

Addresses a scenario when an administrator requires to prove onion ownership to an external CA. Adds a `.well-known` folder to the application server and updates apache and apparmor rules to allow serving arbitrary html files from here. 

## Testing

* Ensure CI is passing (added a test for the debian app code pkg)
* Spin up a staging instance, navigate to the source onion address, drop a file in `/var/www/securedrop/.well-known/` (extension `.html`), and attempt to navigate to that in a tor browser session.

## Deployment

Any special considerations for deployment? Consider both:

1. Upgrading existing production instances.      --- Does not help out with existing production installs fully. An admin will have to re-run the install playbooks to make the appropriate apache changes.
2. New installs. -- Nothing special needs to be done during new deployments.

### If you made changes to the system configuration:

- [ ] [Configuration tests](https://docs.securedrop.org/en/latest/development/testing_configuration_tests.html) pass

### If you made changes to documentation:

- [x] Doc linting passed locally

### CODE REVIEW

**Comment by conorsch** [comment]
This is a candidate for 0.5. Focusing on the Tor apt repo work (e.g. #2441) right now, but let's circle back on this PR and see if we can't get it in.

**Comment by conorsch** [comment]
Reviewing now. Without an EV cert, it's not possible to test the actual workflow, but we can at least verify that the .well-known URLs serve as described in the new documentation, without AppArmor violations.

**Comment by conorsch** [comment]
Works as advertised: 

source-interface-https-onion-validation

Giving the docs a thorough reading to make sure they're as clear as clear can be—an example URL may be warranted.

File: docs/https_source_interface.rst [hunk 1]
```diff
@@ -54,12 +54,18 @@ and to demonstrate control over the Onion URL for your Source Interface.
...
-understand Apache configuration well enough to know how to temporarily
-provision such a page at this specific URL. Once you have received your
-certificate, you should restore the Apache configuration to the stock
-SecureDrop configuration.
+available at a specific URL: ``<onion_url>/.well-known/pki-validation.html``.
+We have support for this work-flow:
+
+.. code:: sh
+
+    # From the admin workstation ssh to the application server
```
Comment (conorsch) [inline docs/https_source_interface.rst]: Nit: capitalization on the machine names.

Comment (conorsch) [reply docs/https_source_interface.rst]: Example URL is already present and looks great.

File: docs/https_source_interface.rst [hunk 2]
```diff
@@ -117,14 +117,14 @@ the configuration: ::
 
 The webserver configuration will be updated to apply the HTTPS settings.
 Confirm that you can access the Source Interface at
-``https://<your_domain>.onion``, and also that the HTTP URL
-``http://<your_domain.onion`` redirects automatically to HTTPS.
+``https://<onion_url>``, and also that the HTTP URL
```
Comment (msheiny) [inline docs/https_source_interface.rst]: 👍 really nice catch

**Pull request merged**